test/intrusive_list_properties.cpp

```diff
@@ -0,0 +1,215 @@
+#include <stdx/intrusive_forward_list.hpp>
+#include <stdx/intrusive_list.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+#include <rapidcheck/catch.h>
+#include <rapidcheck/state.h>
+
+#include <iterator>
+#include <list>
+
+namespace {
+struct int_node {
+    int value{};
+    int_node *prev{};
+    int_node *next{};
+};
+} // namespace
+
+using ListModel = std::list<int>;
+
+template <typename T> struct list_element_deallocator {
+    T list{};
+
+    ~list_element_deallocator() {
+        while (!list.empty()) {
+            delete list.pop_front();
+        }
+    }
+
+    void clear() {
```

Comment:
**suggestion**: Can we DRY out `clear` and the destructor? They do basically the same job?